Assets/Scripts/PlayerCollision.cs

```diff
@@ -26,53 +50,53 @@ private void OnCollisionEnter2D(Collision2D col)
-			              * selfQGB.Power
-			              / opponentQGB.Resistance;
+			Feedback(other, _rigidbody2D.velocity.magnitude + rb.velocity.magnitude);
+			float a1 = Vector2.Angle(rb.velocity, transform.position - other.transform.position);
+			float a2 = Vector2.Angle(_rigidbody2D.velocity, transform.position - other.transform.position);
+			if(a1 > a2)
+			{
+				DoCollision(_rigidbody2D, rb, selfQGB.Power, selfQGB.Resistance, opponentQGB.Power, opponentQGB.Resistance);
+			}
+			else
+			{
+				DoCollision(rb, _rigidbody2D, opponentQGB.Power, opponentQGB.Resistance, selfQGB.Power, selfQGB.Resistance);
+			}
 		}
 	}
-
-	private void OnTriggerEnter2D(Collider2D col)
+	void OnTriggerStay2D(Collider2D col)
 	{
 		GameObject other = col.gameObject;
 		Rigidbody2D rb = other.GetComponent<Rigidbody2D>();
-
+		
 		if (rb != null && other.CompareTag("Player"))
 		{
-			QuantumGyroBlade selfQGB = GetComponent<MovementController>().QGB;
-			QuantumGyroBlade opponentQGB = other.GetComponent<MovementController>().QGB;
-			AudioManager.instance.PlayClangSound();
-			float mag = _rigidbody2D.velocity.magnitude;
-			_rigidbody2D.velocity = (transform.position - other.transform.position)
-			                        * rb.velocity.magnitude
-			                        * CollisionDamping
-			                        * opponentQGB.Power
-			                        / selfQGB.Resistance;
-			rb.velocity = (other.transform.position - transform.position)
-			              * mag
-			              * CollisionDamping
-			              * selfQGB.Power
-			              / opponentQGB.Resistance;
+			_particleSystem.Play();
+			Vector2 dir = gameObject.transform.position - other.transform.position;
+			rb.AddForce(-dir * 20);
+			_rigidbody2D.AddForce(dir * 20);
 		}
 	}
 
-	void Feedback(GameObject other, Vector3 toSpawn, float totalMagnitude)
+	private void DoCollision(Rigidbody2D instigator, Rigidbody2D receiver, float instPow, float instRes, float recPow, float recRes)
+	{
+		Vector2 dir = instigator.transform.position - receiver.transform.position;
+		float recMag = Vector3.Project(instigator.velocity, dir).magnitude * (instPow / recRes);
+		float instMag = Vector3.Project(receiver.velocity, dir).magnitude * (recPow / instRes);
+		//Debug.Log("rec : " + recMag);
+		//Debug.Log("inst: " + instMag);
+		//instigator.velocity =  instigator.velocity + (dir * CollisionDamping * instMag);
+		//receiver.velocity = receiver.velocity + (-dir * CollisionDamping * recMag);
+		instigator.velocity = (dir * globalCollisionMultiplier * instMag);
+		receiver.velocity = (-dir * globalCollisionMultiplier * recMag);
+	}
+
+	void Feedback(GameObject other, float totalMagnitude)
 	{
 		AudioManager.instance.PlayClangSound();
-		_particleSystem.transform.position = toSpawn;
 		Camera.main.GetComponent<CameraController>().ScreenShake(totalMagnitude);
+		_particleSystem.transform.position = (other.transform.position + gameObject.transform.position) / 2;
```

Comment:
This doesn't fix the issue with collisions teleporting player 1 to the point of contact.
In combination with the `OnTriggerStay2D` above, this could result in uncontrollable collisions.

Reply:
The ontriggerstay2d was an attempt to fix when two players spawn on top of eachother or otherwise lerp into one another somehow. The particle system teleportation can be removed seperately if it's still an issue

Reply:
I'm going to try playing with this and see what happens.

Reply:
I just tried out the collision. It's weird. By no means was it uncontrollable, but definitely feels wrong. I'll post a gif to discord in a moment.